response/response.go

```diff
@@ -92,7 +94,19 @@ func (res *Response) WriteTo(w http.ResponseWriter) {
 	res.setStatusCode(w)
 	fmt.Fprint(w, res.Body)
 	if res.keepFunc != nil {
-		res.keepFunc(w)
+		connDone := w.(http.CloseNotifier).CloseNotify()
+		for {
+			select {
+			case <-connDone:
+				return
+			default:
+				keeping := res.keepFunc(w)
```

Comment:
I think goroutine will block in this place, so this goroutine won't return when the connection is done, does this behavior is your expected?

Reply:
Yes, if users want to use go-routine, they have to make sure it would stop after connection done, so what the suggestion is don't use go-routine here.

Reply:
The first branch will be executed? if the user has set `keepFunc`, I don't think there has any case will execute the first branch.

Reply:
Do you mean `<-connDone`?

Reply:
yes

Reply:
https://golang.org/pkg/net/http/#CloseNotifier
Oops, I see that's Deprecated

Reply:
https://golang.org/pkg/net/http/#Request.Context
Should use